tools/automation/tests/__init__.py

```diff
@@ -177,7 +181,7 @@ def discover_tests(args):
 
     CORE_EXCLUSIONS = ['command_modules', '__main__', 'testsdk']
 
-    profile = args.profile.replace('-', '_')
+    profile_namespace = PROFILE_TO_NAMESPACE[args.profile]
```

Comment:
This will crash when a new profile is added that doesn't get added to the PROFILE_TO_NAMESPACE dictionary (which is pretty likely).  If you stick with this, at least catch the error and throw a friendly error to direct the dev to add the profile to the dictionary.

Another alternative would be to have this as part of the standard "shared.py" file in azure.cli.core.profiles. This makes it less likely a profile will be added without this.